src/main/java/com/artipie/RepoPermissions.java

```diff
@@ -61,25 +67,34 @@
     /**
      * Adds or updates repository permission.
      * @param repo Repository name
-     * @param username Username
-     * @param permission Permission name
+     * @param permissions Permissions list
      * @return Completion action
      */
-    CompletionStage<Void> addUpdate(String repo, String username, String permission);
+    CompletionStage<Void> addUpdate(String repo, List<UserPermission> permissions);
```

Comment:
@olenagerasimova I am concerned about `List<UserPermission>`. It makes one think that users are ordered and order does matter. I do not think that this is the case and would suggest to change it to `Collection`. Same goes for `permissions` method return value